recommended.js

```diff
@@ -3,14 +3,23 @@
 const prettier = require('./eslint-plugin-prettier');
 const prettierConfig = require('eslint-config-prettier');
 
-module.exports = {
-  prettierConfig,
-  plugins: {
-    prettier,
-  },
+const plugin = {
+  configs: {},
   rules: {
-    'prettier/prettier': 'error',
-    'arrow-body-style': 'off',
-    'prefer-arrow-callback': 'off',
+    ...prettier.rules,
   },
 };
+
+Object.assign(plugin.configs, {
```

Comment:
Avoid mutations.

Reply:
I'd generally agree, but this is explicitly what ESLint is recommending in the case where you need to reference back to a plugin from a plugin config: https://eslint.org/docs/latest/extend/plugin-migration-flat-config#migrating-configs-for-flat-config

Reply:
Well, that's not always correct or suitable for everyone.

Reply:
Well, that's not always correct or suitable for everyone

Reply:
It feels like now might be a good time to get out of the business of shipping a recommended config that tries to pull in `eslint-config-prettier`.

It should be possible to use `eslint-plugin-prettier` without also needing `eslint-config-prettier` (because you might configure eslint to disable all the conflicting rules some other way).

But the move to flat config makes it so that if we keep the recommended config we have to go and add a peerDependency on `eslint-config-prettier`, rather than relying on the previously kinda weird implicit loading stuff.


@JounQin what would you think to the idea of us removing the recommended config, and instead we document how people should install and enable `eslint-config-prettier` themselves if they want that ruleset.

Reply:
I'm fine with the suggestion.

Reply:
That sounds really good – I found the implicit use of `eslint-config-prettier` without properly defining it as a peer dependency to be strange.

What do you think about leaving `eslint-config-prettier` out for the flat config while keeping the eslintrc config as is for now, to be able to release flat config support in a minor version update?

Then you can decide wether you'd like to removing implicit use of `eslint-config-prettier` in eslintrc config, or if you want to remove the eslintrc config altogether when it's deprecated in ESLint v9.

Reply:
> What do you think about leaving eslint-config-prettier out for the flat config while keeping the eslintrc config as is for now, to be able to release flat config support in a minor version update?

Yes. I think that sounds ideal for this PR. Let's do that.